src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,31 @@
+package seedu.addressbook.data.person;
+
+public class Contact {
+    public String value;
+    protected boolean isPrivate;
+
+    public Contact(boolean isPrivate) {
+        this.isPrivate = isPrivate;
+    }
+
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Contact // instanceof handles nulls
+                && this.value.equals(((Contact) other).value)); // state check
```

Comment:
Note that with this implementation, if an Email and an Address had the same value, they will be treated as equal even though they are different types.